base/docs/basedocs.jl

```diff
@@ -197,6 +197,104 @@ julia> z
 """
 kw"global"
 
+"""
+    =
+
+`=` is the assignment operator.
+* For variables `a` and `b`, `a = b` assigns `a` to `b`, making `a === b` return  `true`.
+* For functions `f(x)`, `f(x) = x` defines a new function constant `f`, or adds a new method to `f` if `f` is already defined; this usage is equivalent to `function f(x); x; end`.
+* `a[i] = v` calls [`setindex!`](@ref)`(a,v,i)`.
+* `a.b = c` calls [`setproperty!`](@ref)`(a,:b,c)`.
+
+# Examples
+Assigning `a` to `b` does not create a copy of `b`; instead use [`copy`](@ref) or [`deepcopy`](@ref).
+
+```jldoctest
+julia> b = [1]; a = b; b[1] = 2; a
+1-element Array{Int64,1}:
+ 2
+
+julia> b = [1]; a = copy(b); b[1] = 2; a
+1-element Array{Int64,1}:
+ 1
+
+```
+Collections passed to functions are also not copied. Thus reassignment within a function can modify the content of an argument. (The names of functions which do this are conventionally suffixed with '!'.)
+```jldoctest
+julia> function f!(x); x[:] .+= 1; end
+f! (generic function with 1 method)
+
+julia> a = [1]; f!(a); a
+1-element Array{Int64,1}:
+ 2
+
+```
+Assignment can operate on multiple variables in parallel, by assigning a [`Tuple`](@ref) to an iterable of at least the same length.
+```jldoctest
+julia> (a, b) = (0, 1) # tuple to tuple
+(0, 1)
+
+julia> a, b = 1:3 # tuple to a longer range (tuples don't always need enclosing parens)
+1:3
+
+julia> a, b
+(1, 2)
+
+```
+Assignment can operate on multiple variables in series, and will return the result of evaluating the right-hand-most variable.
+```jldoctest
+julia> a = [1]; b = [2]; c = [3]; a = b = c
+1-element Array{Int64,1}
+ 3
+
+julia> b[1] = 2; a, b, c
+([2], [2], [2])
+
+```
+A collection cannot be modified by assigning new values to the output of a function. Such modifications can instead be made by assigning new values at the corresponding indices.
```

Comment:
This doesn't seem very clear to me; whether a value was returned from a function is irrelevant. You can indeed modify the object returned by `diag`, it's just that it's a different object from `a`. This example can probably just be removed.